### PR TITLE
fix: `region_peers` returns same region_id for multi logical tables

### LINKED ISSUE #4157: `information_schema.region_peers` returns same `region_id` multiple times
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Table Engine

### Minimal reproduce step

```sql
CREATE TABLE phy (ts timestamp time index, val double) engine=metric with ("physical_metric_table" = "");
CREATE TABLE t1 (ts timestamp time index, val double, host string primary key) engine = metric with ("on_physical_table" = "phy");
CREATE TABLE t2 (ts timestamp time index, job string primary key, val double) engine = metric with ("on_physical_table" = "phy");
select * from information_schema.region_peers;
```

### What did you expect to see?

```
+---------------+---------+-----------+-----------+--------+--------------+
| region_id     | peer_id | peer_addr | is_leader | status | down_seconds |
+---------------+---------+-----------+-----------+--------+--------------+
| 4402341478400 |       0 |           | Yes       | ALIVE  |         NULL |
| 4406636445696 |       0 |           | Yes       | ALIVE  |         NULL |
| 4398046511104 |       0 |           | Yes       | ALIVE  |         NULL |
+---------------+---------+-----------+-----------+--------+--------------+
```

### What did you see instead?

```
+---------------+---------+-----------+-----------+--------+--------------+
| region_id     | peer_id | peer_addr | is_leader | status | down_seconds |
+---------------+---------+-----------+-----------+--------+--------------+
| 4398046511104 |       0 |           | Yes       | ALIVE  |         NULL |
| 4398046511104 |       0 |           | Yes       | ALIVE  |         NULL |
| 4398046511104 |       0 |           | Yes       | ALIVE  |         NULL |
+---------------+---------+-----------+-----------+--------+--------------+
```

### What operating system did you use?

doesn't matter

### What version of GreptimeDB did you use?

main

### Relevant log output and stack trace

_No response_

### PR DESCRIPTION
**Pull request by realtaobo**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
Resolve #4157 

## What's changed and what's your intention?
as title.
image

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Introduced new tables in the `INFORMATION_SCHEMA` database for storing and managing region peer information.
  - Added support for querying region peer data, including distinct counts of region IDs.

- **Bug Fixes**
  - Enhanced the method for adding region peers to handle additional parameters for better data management.

- **Tests**
  - Included comprehensive test cases covering the creation, querying, and deletion of region peer tables in both distributed and standalone environments.

### CODE REVIEW

File: src/catalog/src/information_schema/region_peers.rs [hunk 1]
```diff
@@ -205,18 +205,23 @@ impl InformationSchemaRegionPeersBuilder {
...
-    fn add_region_peers(&mut self, predicates: &Predicates, routes: &[RegionRoute]) {
+    fn add_region_peers(
+        &mut self,
+        predicates: &Predicates,
+        table_id: TableId,
+        routes: &[RegionRoute],
+    ) {
         for route in routes {
-            let region_id = route.region.id.as_u64();
+            let region_id = RegionId::new(table_id, route.region.id.region_number()).as_u64();
```
Comment (realtaobo) [inline src/catalog/src/information_schema/region_peers.rs:224]: Should I keep the `region_id` of the `physical table` unchanged?

**Review by fengjiachun — APPROVED** [review]
LGTM

**Comment by fengjiachun** [comment]
PTAL @WenyXu

**Comment by WenyXu** [comment]
@realtaobo, that is nicely done. Would you like to add a sqlness test for this case?

File: src/catalog/src/information_schema/region_peers.rs [hunk 2]
```diff
@@ -205,18 +205,23 @@ impl InformationSchemaRegionPeersBuilder {
                     table_ids.into_iter().map(|id| (id, vec![])).collect()
                 };
 
-                for routes in table_routes.values() {
-                    self.add_region_peers(&predicates, routes);
+                for (table_id, routes) in table_routes {
```
Comment (WenyXu) [inline src/catalog/src/information_schema/region_peers.rs:208]: nit:
```suggestion
                for (physical_table_id, routes) in table_routes {
```

Comment (realtaobo) [reply src/catalog/src/information_schema/region_peers.rs:208]: There maybe `logical table`, `table_id` is more reasonable?

Comment (fengjiachun) [reply src/catalog/src/information_schema/region_peers.rs:208]: It is logical table @WenyXu

Comment (WenyXu) [reply src/catalog/src/information_schema/region_peers.rs:208]: Good catch, you're right🤩

File: tests/cases/distributed/information_schema/region_peers.result [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    ts timestamp time index,
+) PARTITION ON COLUMNS (a) (
+    a < 10,
+    a >= 10 AND a < 20,
+    a >= 20,
+);
+
+Affected Rows: 0
+
+SELECT region_id,is_leader,status,down_seconds FROM region_peers ORDER BY region_id;
```
Comment (WenyXu) [inline tests/cases/distributed/information_schema/region_peers.result:38]: Oh, my bad. It's hard to test for these cases. The RegionId may be changed due to the execution being in a different order. How about changing SQL to `SELECT count(distinct region_id) FROM region_peers ORDER BY region_id;`, then we can move this test into `common` dir.

**Review by WenyXu — APPROVED** [review]

**Comment by WenyXu** [comment]
Thanks!

**Pull request merged**